### PR TITLE
added loading screen

### PR DESCRIPTION
**Pull request by okushama**
screen subtitles populate via categories from the mod language file
renders two mobs from the tropicraft egg list (minus some irregularly
sized ones)
have tweaked water mob texture selection code to account for this

### CODE REVIEW

File: src/main/java/net/tropicraft/core/client/TropicraftLoadingListener.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+				GlStateManager.color(1f, 1f, 1f, 1f);
+				
+						
+			}
+		}else {
+			this.screenReassign = true;
+		}
+	}
+	
+	public ItemStack stack(Item i) {
```
Comment (tterrag1098) [inline src/main/java/net/tropicraft/core/client/TropicraftLoadingListener.java]: I don't see the point of these methods

File: src/main/java/net/tropicraft/core/client/TropicraftLoadingListener.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        VertexBuffer vertexbuffer = tessellator.getBuffer();
+        TropicraftRenderUtils.bindTextureGui(screenBackground);
+        GlStateManager.color(0.9F, 0.9F, 0.9F, 1.0F);
+        float f = 128.0F;
+        vertexbuffer.begin(7, DefaultVertexFormats.POSITION_TEX_COLOR);
+     
+        vertexbuffer.pos(0,     height, 0)	.tex(0,         height / f)	.color(0.5f, 0.5f, 0.5f, 1f).endVertex();
+        vertexbuffer.pos(width, height, 0)	.tex(width / f, height / f)	.color(0.5f, 0.5f, 0.5f, 1f).endVertex();
+        vertexbuffer.pos(width, 0,      0)	.tex(width / f, 0)			.color(0.5f, 0.5f, 0.5f, 1f).endVertex();
+        vertexbuffer.pos(0,     0,      0)	.tex(0,         0)			.color(0.5f, 0.5f, 0.5f, 1f).endVertex();
```
Comment (tterrag1098) [inline src/main/java/net/tropicraft/core/client/TropicraftLoadingListener.java]: Weird indents here and below

**Review by tterrag1098 — CHANGES_REQUESTED** [review]
Also, couldn't the second two backgrounds be 16x16? They tile anyways, and as far as I can see there isn't any variation in the pattern.

File: src/main/java/net/tropicraft/core/client/TropicraftLoadingListener.java (hunk 1 above)
Comment (okushama) [reply src/main/java/net/tropicraft/core/client/TropicraftLoadingListener.java]: saves a 'new item' being typed every time i need an itemstack, over a lifetime i could save a few hours just by not typing new ItemStack

**Comment by okushama** [comment]
going to implement a basic level of IMC as Twilight Forest wants to use a custom loader for entry/exit of their dimension so we need to exclude their dimension from farewell overrides in cases where somebody travels directly from one to the other

**Pull request closed without merging**